bin/cli.js

```diff
@@ -148,33 +147,40 @@ const setRequestHeaders = () => {
   addUserAgentHeader('HubSpot CLI', pkg.version);
 };
 
-const loadConfigMiddleware = async options => {
-  // Load the new config and check for the conflicting config flag
-  if (configFileExists(true)) {
-    loadConfig('', options);
+const NO_CONFIG_VALIDATION = {
+  init: { skip: true },
+  auth: { skip: true },
+};
 
-    if (options.config) {
-      logger.error(
-        i18n(`${i18nKey}.loadConfigMiddleware.configFileExists`, {
-          configPath: getConfigPath(),
-        })
-      );
+const loadConfigMiddleware = async options => {
+  const maybeValidateConfig = () => {
+    const shouldValidate = options._.every(
+      command =>
+        !(NO_CONFIG_VALIDATION[command] && NO_CONFIG_VALIDATION[command].skip)
+    );
+    if (shouldValidate && !validateConfig()) {
       process.exit(EXIT_CODES.ERROR);
     }
-    return;
-  }
+  };
 
-  // We need to load the config when options.config exists,
-  // so that getAccountIdFromConfig() in injectAccountIdMiddleware reads from the right config
-  if (options.config && fs.existsSync(options.config)) {
+  // Load the new config and check for the conflicting config flag
+  if (configFileExists(true) && options.config) {
+    logger.error(
+      i18n(`${i18nKey}.loadConfigMiddleware.configFileExists`, {
+        configPath: getConfigPath(),
+      })
+    );
+    process.exit(EXIT_CODES.ERROR);
+  } else {
```

Comment:
This could be cleaned up to `} else if (options._.includes('init')) {`. I also think you can get rid of the comments here since they're less relevant after the latest iteration.

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1307/commits/24a38b52cc51d9d07959ff79fbb02e5078eb174b